eXtendingHEEP.md

```diff
@@ -0,0 +1,309 @@
+        file_type: systemVerilogSource
+    
+    tb:
+        depend:
+        - x-heep::tb-utils
+        files:
+        - tb/testharness.sv   # Your modified testharness
+        - hw/vendor/esl-epfl_x-heep/tb/tb_top.sv
+        file_type: systemVerilogSource
+
+    # Scripts for hooks
+    pre_build_uartdpi:
+        files:
+        - scripts/sim/compile_uart_dpi.sh  # Your modified scripts
+        file_type: user
+
+    pre_build_remote_bitbang:
+        files:
+        - scripts/sim/compile_remote_bitbang.sh  # Your modified scripts
+        file_type: user
+
+    pre_patch_modelsim_Makefile:
+        files:
+        - hw/vendor/esl-epfl_x-heep/scripts/sim/modelsim/patch_modelsim_Makefile.py
+        file_type: user
+
+    parameters:
+    PULP_XPULP:
+        datatype: int
+        paramtype: vlogparam
+        default: 0
+    JTAG_DPI:
+        datatype: int
+        paramtype: vlogparam
+        default: 0
+    USE_EXTERNAL_DEVICE_EXAMPLE:
+        datatype: bool
+        paramtype: vlogdefine
+        default: false
+    USE_UPF:
+        datatype: bool
+        paramtype: vlogdefine
+        default: false
+
+    scripts:
+    pre_build_remote_bitbang:
+        cmd:
+        - sh
+        - ../../../scripts/sim/compile_remote_bitbang.sh
+    pre_build_uartdpi:
+        cmd:
+        - sh
+        - ../../../scripts/sim/compile_uart_dpi.sh
+    pre_patch_modelsim_Makefile:
+        cmd:
+        - python
+        - ../../../hw/vendor/esl-epfl_x-heep/scripts/sim/modelsim/patch_modelsim_Makefile.py
+
+    targets:
+    default: &default_target
+        filesets:
+        - files_rtl_generic
+
+    sim:
+        <<: *default_target
+        default_tool: modelsim
+        filesets_append:
+        - tb
+        - tool_modelsim? (pre_build_remote_bitbang)
+        - tool_modelsim? (pre_build_uartdpi)
+        - tool_modelsim? (pre_patch_modelsim_Makefile)
+        toplevel:
+        - tb_top
+        hooks:
+        pre_build:
+            - tool_modelsim? (pre_build_uartdpi)
+            - tool_modelsim? (pre_build_remote_bitbang)
+            - tool_modelsim? (pre_patch_modelsim_Makefile) # this is required by Questa 2020 on
+        parameters:
+        - PULP_XPULP=0
+        - use_jtag_dpi? (JTAG_DPI=1)
+        - "!use_jtag_dpi? (JTAG_DPI=0)"
+        - use_external_device_example? (USE_EXTERNAL_DEVICE_EXAMPLE=true)
+        - use_upf? (USE_UPF=true)
+        tools:
+        modelsim:
+            vlog_options:
+            - -override_timescale 1ns/1ps
+            - -suppress vlog-2583
+            - -suppress vlog-2577
+            - -pedanticerrors
+            - -define MODELSIM
+            vsim_options:
+            - -sv_lib ../../../hw/vendor/esl-epfl_x-heep/hw/vendor/lowrisc_opentitan/hw/dv/dpi/uartdpi/uartdpi
+            - -sv_lib ../../../hw/vendor/esl-epfl_x-heep/hw/vendor/pulp_platform_pulpissimo/rtl/tb/remote_bitbang/librbs
+```
+
+</details>
+
+
+## Vendorizing X-HEEP
```

Comment:
(I think that) This section needs to be the first one. For instance, if you go to line 40, we are already dealing with vectorization

Reply:
You are right. It was a mistake.